src/config.rs

```diff
@@ -146,6 +148,16 @@ pub(crate) struct MqttConfig {
 
     #[serde(default)]
     pub(crate) client_auth: Option<(std::path::PathBuf, std::path::PathBuf)>,
+
+    #[serde(default)]
+    pub(crate) discovery: Option<MqttDiscoveryConfig>,
+}
+
+#[derive(Debug, Deserialize, Clone, Validate)]
+pub(crate) struct MqttDiscoveryConfig {
+    pub(crate) topic: String,
+
+    pub(crate) features: HashSet<Discoveries>,
```

Comment:
Thought I'd explain some changes to help with future PRs

Here I use a `HashSet` over `Vec` because that will ensure all entries only appear once.

I also use the enum `Discoveries` because that will only permit certain values. It also will give use good error messages like this:
```
Error: Failed to parse the "target/test.toml" config file

Caused by:
    TOML parse error at line 18, column 60
       |
    18 |     features = ["camera", "led", "ir", "motion", "reboot", "pts"]
       |                                                            ^^^^^
    unknown variant `pts`, expected one of `Floodlight`, `Camera`, `Motion`, `Led`, `Ir`, `Reboot`, `Pt`
```

Reply:
Good idea. Neat that the library handles all that for you automatically and the type system helps!